standard/placement.lua

```diff
@@ -110,127 +29,152 @@ local placeColorClass = {
 	['14'] = 'placement-blue',
 	['15'] = 'placement-blue',
 	['16'] = 'placement-blue',
-	['32'] = 'placement-darkblue',
-	['256'] = 'placement-darkgrey',
+	['17'] = 'placement-darkblue',
+	['129'] = 'placement-darkgrey',
+	['q'] = 'placement-win',
 	['w'] = 'placement-win',
+	['d'] = 'placement-draw',
 	['l'] = 'placement-lose',
 	['dq'] = 'placement-lose',
 	['dnp'] = 'placement-dnp',
 	['proceeded'] = 'placement-up',
 	['stay'] = 'placement-stay',
 	['relegated'] = 'placement-down',
+	['div'] = 'placement-darkgrey',
 }
 
-local textColor = {
-	-- White is the default color. Only overrides will need to be entered here.
-	['dnp'] = 'black',
+local customSorts = {
+	['w'] = '1025',
+	['proceeded'] = '1026',
+	['q'] = '1027',
+	['d'] = '1028',
+	['stay'] = '2019',
+	['l'] = '1030',
+	['relegated'] = '1031',
+	['dnp'] = '1032',
+	['dq'] = '1033',
+	['div'] = '1034',
+	[''] = '1035',
 }
 
-local skipShadow = {
-	-- Default is false
-	['dnp'] = true,
+local prizepoolClasses = {
+	'background-color-first-place',
+	'background-color-second-place',
+	'background-color-third-place',
+	'background-color-fourth-place',
+	w = 'bg-win',
+	q = 'bg-win',
+	d = 'bg-draw',
+	l = 'bg-lose',
+	dq = 'bg-dq',
 }
 
-function Placement._placement(args)
-	-- Defaults
-	local textColorCell = 'white'
-	local skipShadowCell = false
-
-	-- Attempt to split the placement into two parts and store it in a table
-	local placement = mw.text.split(tostring(args.placement), '-', true)
-
-	-- If the table has 2 parts, the placement has a start and end number
-	local text
-	local diff
-	if tonumber(placement[1]) == nil then
-		text = string.upper(placement[1])
-		diff = 0
-	elseif table.maxn(placement) == 2 then
-		if ordinalSuffix[placement[2]] then
-			text = placement[1] .. '&nbsp;-&nbsp;' .. placement[2] .. ordinalSuffix[placement[2]]
-		else
-			text = placement[1] .. '&nbsp;-&nbsp;' .. placement[2] .. ordinalSuffix[string.sub(placement[2], -1)]
-		end
-		diff = (tonumber(placement[2]) or 1000) - tonumber(placement[1])
+---Processes a placement text input into raw data.
+---Returned table will not always contain every key.
+---@param placement string?
+---@return table
+function Placement.raw(placement)
+	local raw = {}
+
+	-- Nil check on input and split placement if joint
+	raw.placement = mw.text.split(string.lower(placement or ''), '-', true)
+
+	-- Identify appropriate background class
+	if raw.placement[1] == '3' and raw.placement[2] then
+		raw.backgroundClass = placementClasses['d']
+	elseif placementClasses[raw.placement[1]] then
+		raw.backgroundClass = placementClasses[raw.placement[1]]
+	elseif tonumber(raw.placement[1]) <= 128 then
```

Comment:
can cause issues if we have a non number, then it will try to compare nil with a number (i.e. error)